### DISCUSSION TITLE
iOS app stops with a black screen when device is rotated during startup

### DISCUSSION THREAD

**Discussion by kasparv** [airsdk/Adobe-Runtime-Support#1913] [Q&A]
For the iOS AIR app I have the following settings in the app descriptor:

```
<aspectRatio>landscape</aspectRatio>
<autoOrients>true</autoOrients>
<fullScreen>true</fullScreen>
```

When the device is being rotated during app startup, the screen goes black after showing launch screen and the app won't proceed to the AIR/Flash content.
Is this a known issue? 
Are there any workarounds besides setting autoOrients to false? I would like the customers to have a choice which way in landscape they can keep the device to use the app.

The launch screen itself is done in Xcode and working fine, the app is already on the App Store but it's an annoying bug that I'd like to fix...

**Comment by GeneralVimes** [comment]
I confirm - after reading your post I checked my games https://apps.apple.com/in/app/farm-and-mine-deep-idle-ld48/id1564694016 (built with AIR 33.1.1.821) and the bug is present there. To reproduce the device must be rotated at the moment Harman splash is shown.

**Reply by GeneralVimes** [reply]
Also it was possible to reproduce the bug with an earlier made game https://apps.apple.com/in/developer/alexey-izvalov/id1437172357 built with a paid tier AIR SDK 33.1.1.476
The device must be rotated right after the start

**Comment by FliplineStudios** [comment]
I'm seeing the same thing with our game built with 33.1.1.743: https://apps.apple.com/app/id1591627945?ls=1
Same setup with landscape, fullscreen, autoOrients, on the paid tier with a custom LaunchScreen Storyboard.  Also seeing the same for another game build with 33.1.1.476.

However I am **not** seeing this behavior with an older game built with 33.1.1.63 (https://apps.apple.com/app/id1504952058?ls=1), which was before Harman changed their own splash screen to use a launch Storyboard in the SDK.  This app with .63 is set up the same way, using our own custom LaunchScreen Storyboard on the paid tier.  Based on this, I'm guessing it's something within the Harman launch storyboard handling (even when using paid-tier custom storyboards) in these newer releases that's causing this problem.

Also somewhat unrelated, but after looking back at this older app, I forgot how much faster iOS AIR apps used to launch with Storyboards, and it was only after the Harman splash mechanism was changed to Storyboards that it added quite a bit of delay...

**Comment by ajwfrost** [comment]
Interesting - thanks for raising this one! We'll look into it, it definitely seems related to the launch storyboard mechanism...

**Comment by FliplineStudios** [comment]
@ajwfrost Any update on fixing this issue with rotating during launch in a future update?  

We've been getting more emails from our users having this problem with our latest apps.  Especially since our launch storyboard has a "Loading..." graphic that's oriented for landscape, so users are naturally rotating their phones when that storyboard is being shown and then getting stuck with the black screen immediately after...

**Reply by ajwfrost** [reply]
Hi - my colleague who was looking at this one isn't in today but I just checked the latest notes on this, sounds like we can reproduce it with the applications mentioned above but we've not been able to craft our own application that reproduces it. Having just checked it myself I also don't get this happening with a basic app, I'm wondering whether it's perhaps Stage3D-related..

Can I check whether everyone who's seen this is using Stage3D? and Starling? Any other info (or even a simple 'hello world' style app - swf + xml - that reproduces the problem) would be handy.

To confirm our test case, it's just a normal swf with some text field, with those three tags from the original post added to the descriptor file. When it launches, the Harman splash screen is shown in landscape, and we then rotate the device so that it's shown upside down. Once the actual SWF content appears, it then is rotated and displays normally for us.

thanks

**Reply by GeneralVimes** [reply]
Yes, my game uses Starling

**Reply by ajwfrost** [reply]
Hmm... so a normal stage3d app still works for us, as does a basic starling app.

If anyone can put together a simple test case that reproduces this where we can then package it up from the SWF + XML file, it would be very helpful!

thanks

**Reply by FliplineStudios** [reply]
We're not using Stage3D or Starling, it's just a regular display list app in GPU mode, though it seems to happen in any render mode. 

> When it launches, the Harman splash screen is shown in landscape, and we then rotate the device so that it's shown upside down. Once the actual SWF content appears, it then is rotated and displays normally for us.

Sorry, maybe I wasn't clear on what we're seeing -- in our case, we're starting out holding the phone in **portrait** mode as the landscape-only app launches.  The custom launch storyboard correctly appears as landscape/sideways while I'm still holding the phone upright, but now if I turn the phone from portrait to landscape as this launch storyboard is showing, it will then show a black screen once the app starts up (though you can hear its audio in the background).  

> we then rotate the device so that it's shown upside down

I don't think upside-down is a factor in this for us, it's when physically turning the phone from portrait to landscape while the app is starting up that it happens, causing the app's content to not be visible after the storyboard is removed.

**Reply by kasparv** [reply]
> Hi - my colleague who was looking at this one isn't in today but I just checked the latest notes on this, sounds like we can reproduce it with the applications mentioned above but we've not been able to craft our own application that reproduces it. Having just checked it myself I also don't get this happening with a basic app, I'm wondering whether it's perhaps Stage3D-related..
> 
> Can I check whether everyone who's seen this is using Stage3D? and Starling? Any other info (or even a simple 'hello world' style app - swf + xml - that reproduces the problem) would be handy.
> 
> To confirm our test case, it's just a normal swf with some text field, with those three tags from the original post added to the descriptor file. When it launches, the Harman splash screen is shown in landscape, and we then rotate the device so that it's shown upside down. Once the actual SWF content appears, it then is rotated and displays normally for us.
> 
> thanks

In our case it's Flex app, no Starling or Stage3D.

**Reply by ajwfrost** [reply]
Thanks - still no reproduction here which is odd.. @FliplineStudios we have tried that sequence too i.e. start holding the phone portrait, then rotate the phone when the splash screen appears.

One other question though: is this only happening if it's a production license? I'm assuming that we'd be able to reproduce it when using the built-in splash screen for non-commercial users, because it should be the same thing i.e. a storyboard file added to the info.plist .. if you take out your adt license file and re-package does the same thing happen when you have the Harman/AIR launch storyboard?

thanks

**Reply by FliplineStudios** [reply]
I don't have all of my iOS devices with me at the moment, but strangely I can't seem to trigger the problem on this iPhone X, but with an iPad I can get it to happen about 75% of the time -- and weirdly it triggers more often when going from portrait to upside-down landscape on this device.  I'll see if I can get some better test cases, and try without the production license too...

**Reply by FliplineStudios** [reply]
Quick update: It's also happening using the free tier, when I remove the license and custom launch storyboard and default to the Harman splash storyboard.  It may be the timing of when you rotate the device has some effect on whether this problem occurs.  I should note our app is around 30MB with lots of content (thousands of MovieClips of vector art), so it does sit on the splash storyboard for a couple seconds, giving a wider window of time where we can rotate it and see this problem happening (in case your test project starts up really fast and you have a smaller window of time to try to rotate it)

**Reply by itlancer** [reply]
@ajwfrost 
Issue still exists with latest AIR 51.0.1.4. Can be reproduced with any iOS device (iPhones and iPads) with any iOS version. Happens in ~70% launches. Sometimes it could be reproduced only in "one way rotation". I mean lanscape->portrait or portrait->lanscape. Storyboard should be used by app and auto-rotate should be enabled on device in OS settigns.

I noticed there is no such issue with AIR 51 BETA (51.0.0.2). Seems "BETA version SDK" popup invalidate something and all works correctly.

I notice that sometimes it could be "fixed" after launch by some actions may be related to `StageText` focusing or something like that. Application could start to render normally again.

**Reply by itlancer** [reply]
@ajwfrost 
Issue still exists with latest AIR 51.1.3.7.

And again, it works fine with AIR 51.2.0.1 BETA. Seems "BETA version SDK" popup invalidate something and all works correctly.

**Reply by itlancer** [reply]
@ajwfrost
Issue still exists with latest AIR 51.2.2.6.

**Reply by itlancer** [reply]
Issue still exists with latest AIR 51.3.2.2.

**Reply by ajwfrost** [reply]
I don't suppose if you're able to get an app to reproduce it, you could build the app with the symbol export done via the https://airsdk.harman.com/support#ipa_symbols mechanism, and then send us the symbol data and the IPS crash log? We just had another quick attempt to see if we could reproduce this but still no joy..

thanks
